src/ServiceControl/Hosting/Commands/CustomLivetimes/PersisterInitializingWindowsServiceLifetime.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Hosting.WindowsServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ServiceControl.Persistence;
+
+sealed class PersisterInitializingWindowsServiceLifetime : WindowsServiceLifetime, IHostLifetime
+{
+    readonly IPersistenceLifecycle persistenceLifecycle;
+
+    public PersisterInitializingWindowsServiceLifetime(
+        IPersistenceLifecycle persistenceLifecycle,
```

Comment:
This is very subjective but I would have added all be base constructor parameters first and then the lifecycle one as an addition last.

Reply:
Did that in the last commit. I personally favor the previous as that allows all types + invocation of the base class to be on the same line.

What do you think?